app/services/attempts_api/redis_client.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module AttemptsApi
+  class RedisClient
+    def write_event(event_key:, jwe:, timestamp:, issuer:)
+      key = key(timestamp, issuer)
+      REDIS_ATTEMPTS_API_POOL.with do |client|
+        client.hset(key, event_key, jwe)
+        client.expire(key, IdentityConfig.store.attempts_api_event_ttl_seconds)
+      end
+    end
+
+    def read_events(timestamp:, issuer:, batch_size: 5000)
+      key = key(timestamp, issuer)
+      events = {}
+      REDIS_ATTEMPTS_API_POOL.with do |client|
+        client.hscan_each(key, count: batch_size) do |k, v|
+          events[k] = v
+        end
+      end
+      events
+    end
+
+    def remove_events(timestamp:, issuer:)
```

Comment:
[question] are we going to be deleting events? don't they just expire at the designated ttl?

Reply:
My understanding of the poll spec is that individual events need to be acknowledged by their JTI, so we would be deleting by that instead of by time?

Reply:
With the way we store them currently, they would expire, yeah. This method probably doesn't need to be included at this point, so I'll remove it.